Sources/SQLKit/Query/SQLConstraint.swift

```diff
@@ -17,9 +17,27 @@ public struct SQLConstraint: SQLExpression {
     public func serialize(to serializer: inout SQLSerializer) {
         if let name = self.name {
             serializer.write("CONSTRAINT ")
-            name.serialize(to: &serializer)
+            if let identifier = (name as? SQLIdentifier)?.string {
+                let normalizedName = serializer.dialect.normalizeSQLConstraintIdentifier(identifier)
+                SQLIdentifier(normalizedName).serialize(to: &serializer)
+            } else {
+                name.serialize(to: &serializer)
+            }
             serializer.write(" ")
         }
         self.algorithm.serialize(to: &serializer)
     }
 }
+
+extension SQLDialect {
+    public func normalizeSQLConstraintIdentifier(_ identifier: String) -> String {
```

Comment:
Should this be a protocol requirement with a default implementation so that the dialect can choose what it wants to do?